Imperia Frankfurt Parliament

Danarca opened the pull request.

CattyCatto committed:
Added 00_german_unification.txt to journal_entries

CattyCatto committed:
Add files via upload

CattyCatto committed:
Re-encoded txt files to UTF-8 BOM

CattyCatto committed:
Add files via upload

CattyCatto committed:
Re-encoded txt files to UTF-8 BOM

CattyCatto committed:
Delete common/journal_entries/babylon_Frankfurt_Conference.txt

CattyCatto committed:
Add files via upload

CattyCatto committed:
Re-encoded txt files to UTF-8 BOM

Danarca committed:
First 2 (+ the optional republican form) votes are done

Danarca committed:
More - 1.5 button sets done

CattyCatto committed:
Add files via upload

CattyCatto committed:
Re-encoded txt files to UTF-8 BOM

CattyCatto committed:
Delete common/journal_entries/00_german_unification.txt

not needed for conference.

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
Frankfurt Conference Imperia rebrand complete (?)

Danarca committed:
German Solution buttons

Danarca committed:
Morning routine

Danarca committed:
Voting for the Gutter king completed + localization

Danarca committed:
Start of Gutter King consolidation

Danarca committed:
Further work on consolidation phase

Danarca committed:
Almost done with consolidation

Danarca committed:
Germany now forms if consolidation is successful, and uses the black-red-gold flag

Danarca committed:
[Rename] conference -> parliament

Danarca committed:
Frankfurt Parliament QOL

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
state_is_gutter_german trigger added

Danarca committed:
Gutter King returns to old laws after failure

Danarca committed:
Performance enhancing code

Danarca committed:
Missed the buttons

Danarca committed:
QOL + Gelre is not German (whoops)

Danarca committed:
Consolidating Gutter Germany will give claims on the rest of the claimed lands + Alsace-Lorraine is claimed by Greater Austria

Danarca committed:
Frankfurt Parliament Release Candidate 1

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
Fix for premature variable clean

Danarca committed:
Possibility for Austria shattering in case of Greater Germany

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
AI now considers friendly majors voting patterns

Danarca committed:
3-step swaying process (!); Gutter Crown negative modifier

Danarca committed:
Fixes, code optimisation, QOL and more

Danarca committed:
QOL, fixes, balance

Danarca committed:
Changes as per Fallonars feedback pt. 1

*6 sway modifiers -> 4 ones
*Greater Germany now spawns Rhineland
*Voting period -> 16 weeks
*Greater Austria no longer includes true neutral countries (looks at the Swiss)
*Tax waste for gutter crown modifier
*Gutter crown legitimacy changed from -35 to -30
*Infamy decay for gutter crown modifier

Danarca committed:
Start of AI restrictions on trying to form Germany while the parliament is underways

Danarca committed:
Create 00_major_formables.txt

Danarca committed:
Can no longer launch leadership plays or form Germany if the thing is ongoing

Danarca committed:
Work on clarification

Danarca committed:
Take 1 of explanations/clarifications

Danarca committed:
Further balancing + revolutionary germany modifier

Danarca committed:
Words and security railing

Danarca committed:
Create 00_german_unification.txt

Danarca committed:
Bug-fixes and QOL

Danarca committed:
Vote counts are displayed; better consolidation initialisation

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
Vote count variable cleanup

Danarca committed:
Reduced flip-flopping votes

Danarca committed:
non-functional, experimental parliamentary club influence

Danarca committed:
Division by 10

Danarca committed:
First 2 parliamentary club events

Danarca committed:
4/6(?) club events done; influence system functionally plugged in

Danarca committed:
Last events and fixing it all up; +QOL

Danarca committed:
Word salad

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
Button QoL + NO flip-flopping; GPs will never accept subjugation

Danarca committed:
Better localization

Danarca committed:
Decision to give up Gutter Crown; fails the journal entry

Danarca committed:
Civil war work [WONT BOOT]

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-frankfurt-conference

Danarca committed:
Work on german civil war

Danarca committed:
Finally works

Danarca committed:
Vote result toasts; moved Frankfurt custom_loc to own file

Danarca committed:
Authority more streamlined

Danarca committed:
More variable cleanup

Danarca committed:
Notifications for consolidation scenarios

Danarca committed:
Refactorings

Danarca committed:
Fixes

Danarca committed:
QoL, Fixes, variable cleanup

HiddeLekanne commented on common/customizable_localization/imperia_frankfurt_parliament_localisation.txt:
```diff
@@ -0,0 +1,395 @@
+	}
+}
+
+imperia_frankfurt_parliament_current_vote = {
```
You should mention that you're (ab)using custom localisation to completely change the shown text as if it's a totally new event. This is not a normal use of custom localisation. Normally you do an event chain.

HiddeLekanne commented on common/customizable_localization/imperia_frankfurt_parliament_localisation.txt:
```diff
@@ -0,0 +1,395 @@
+	}
+}
+
+imperia_frankfurt_parliament_current_vote_result = {
```
You should mention that you're (ab)using custom localisation to completely change the shown text as if it's a totally new event. This is not a normal use of custom localisation. Normally you do an event chain.

HiddeLekanne commented on common/diplomatic_plays/00_diplomatic_plays.txt:
I don 't see any changes besides formatting, can you reverse it? For merging with vanilla this is not helpful.

HiddeLekanne commented on common/diplomatic_plays/imperia_diplomatic_plays.txt:
```diff
@@ -106,3 +106,23 @@ dp_imperia_great_war = {
+	allow_negotiated_peace = no
+	mirror_war_goal = yes
+	initiator_can_add_war_goals = no
+    target_can_add_war_goals = no
```
formatting

HiddeLekanne commented on common/diplomatic_plays/imperia_diplomatic_plays.txt:
```diff
@@ -106,3 +106,23 @@ dp_imperia_great_war = {
 	}
 }
+
+dp_imperia_gutter_german_revolution = {
```
Is this an revolution? If it is, why not make it a mechanical revolution. If it's not, call it something else.

HiddeLekanne commented on common/flag_definitions/imperia_flag_definitions.txt:
```diff
@@ -287,6 +287,18 @@ KUR = { #Kurdistan
 	}
 }
 
+GER = {
```
Wait this works additively? Why don't we do that with all our flags? xd

Danarca replied:
I'm not sure exactly what you're talking about, I'm taking the same approach as has been done elsewhere in the file?

HiddeLekanne replied:
I mean that you can just add flags to already existing flag definitions without overwriting them

HiddeLekanne commented on common/journal_entries/00_german_unification.txt:
is the only change that percent is decreased from 0.85 to 0.75? then don't format, it's not helpful for vanilla comparison.

Danarca replied:
No there's some hidden_triggers up above., line 116 and 130

Danarca committed:
Theoretical German Major blurb for the Gutter King

Danarca committed:
Theoretical vote tally counts (updated weekly)

Danarca committed:
German Major blub for Gutter King and German Princes

Danarca committed:
New sway menu

Danarca committed:
Proper localisation

Danarca committed:
Notify on sway from any country if you're involved

Danarca committed:
Clean up sway variables

Danarca commented:
- German Opponent now more prominent in JE text
- New sway system (!!!)
- Notify on enemy sway (or if you're a German Minor (why?) you get notifications from both sides)
- Current tally updated weekly during voting

HiddeLekanne commented on common/journal_entries/imperia_frankfurt_parliament.txt:
```diff
@@ -0,0 +1,984 @@
+			}
+			if = {
+				limit = {
+					this = p:x11C0A0.owner
```
Comment that this is Frankfurt.

HiddeLekanne replied:
Also this will probably break when the map gets updated. But we'll see

HiddeLekanne commented on common/on_actions/imperia_on_capitulation.txt:
```diff
@@ -0,0 +1,58 @@
+		# If the Gutter King lost
+		if = {
+			limit = {
+				has_variable = accepted_gutter_crown
+				NOT = {
+					has_journal_entry = je_imperia_frankfurt_german_consolidation_gutter_king
+				}
+				has_global_variable = frankfurt_parliament_civil_war_happened
```
One of these variables need to be reset in this effect, otherwise this might trigger again when the gutter king loses again to something completely unrelated.

Danarca replied:
Both of them gets cleaned up in `imperia_frankfurt_parliament_clean_variables`, which fires from on_monthly_pulse_country on_action line 134. Conditions are basically that no more Gutter Germany shenanigans is happening.

HiddeLekanne commented on common/on_actions/imperia_on_monthtly_pulse.txt:
```diff
@@ -129,6 +129,36 @@
+			limit = {
+				imperia_frankfurt_parliament_is_in_session = no
+				# At least one country has one of the following variables...
+				any_country = {
```
Why do we check every country for every country? Can't we just check the current country only?

HiddeLekanne replied:
Also why can't we clean up the variables when the journal entry becomes invalid/completes or whatever.

Danarca replied:
The reason I'm not cleaning up per JE is that many of the variables are used down the line. Although I suppose I could split up the clean effect, so the JE-super specific variables gets cleaned.. 
But good point on the any_country in an every_country, I'll perform a fix for that

HiddeLekanne commented on common/script_values/imperia_frankfurt_parliament_values.txt:
```diff
@@ -0,0 +1,1334 @@
+	}
+}
+
+imperia_frankfurt_parliament_greater_germany_ai_willingness_value = {
```
This needs to be communicated a lot more to the player.

HiddeLekanne commented on common/script_values/imperia_frankfurt_parliament_values.txt:
```diff
@@ -0,0 +1,1334 @@
+	}
+}
+
+imperia_frankfurt_parliament_lesser_germany_ai_willingness_value = {
```
This needs to be communicated a lot more to the player.

HiddeLekanne commented on common/script_values/imperia_frankfurt_parliament_values.txt:
```diff
@@ -0,0 +1,1334 @@
+	}
+}
+
+imperia_frankfurt_parliament_greater_austria_ai_willingness_value = {
```
This needs to be communicated a lot more to the player.

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1885 @@
+	}
+}
+
+je_frankfurt_parliament_choose_club_button = {
```
Is this tested in multiplayer?

Danarca replied:
Nope, I'll set up a playdate with Lou

Danarca replied:
Alright, our date just ended :p It all seems to work in multiplayer. I also found 2 other errors, that I'll get around to fixing

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1885 @@
+			}
+			add = 75
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:cordial
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 20
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:amicable
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 50
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:friendly
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_republic
+				}
+			}
+			add = 80
+		}
```
How do these even work? The conditions are cyclical... Is this only for players?

Danarca replied:
The higher the relations with a major german power that voted for x option, the more likely they are to vote for said option.
If they dont have domestic support through the clubs however they wont vote for it at all. And once they DO have voting support they will swing their vote hard.
This is to prevent flip-flopping, although I did notice that a few votes were constantly swinging around after I added the "live" vote view.. So Im not sure whats up there.

HiddeLekanne replied:
> The higher the relations with a major german power that voted for x option, the more likely they are to vote for said option. If they dont have domestic support through the clubs however they wont vote for it at all. And once they DO have voting support they will swing their vote hard. This is to prevent flip-flopping, although I did notice that a few votes were constantly swinging around after I added the "live" vote view.. So Im not sure whats up there.

So their ai chance depends on the other german majors, which depends on other german majors etc.

Danarca replied:
Hmm, so perhaps these should not be relevant for major powers, only for lesser german princes?

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1885 @@
+				imperia_frankfurt_parliament_law_is_preferred_by_dominating_left_ai_trigger = no
+				has_variable = frankfurt_parliament_voted_monarchy
+			}
+			multiply = 0
```
Seems harsh lol,

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1885 @@
+			}
+			add = 75
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:cordial
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_monarchy
+				}
+			}
+			add = 20
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:amicable
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_monarchy
+				}
+			}
+			add = 50
+		}
+		modifier = {
+			trigger = {
+				any_country = {
+					has_journal_entry = je_imperia_frankfurt_parliament
+					relations:root >= relations_threshold:friendly
+					country_rank >= rank_value:major_power
+					has_variable = frankfurt_parliament_voted_monarchy
+				}
+			}
+			add = 80
+		}
```
How do these even work? The conditions are cyclical... Is this only for players?

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1885 @@
+				}
+				has_variable = frankfurt_parliament_voted_republic
+			}
+			multiply = 0
```
again seems harsh

HiddeLekanne reviewed (changes requested):
Partial review

Danarca committed:
Revert 00_diplomatic_plays.txt to Core

Danarca committed:
Re-adding the session check in dp_leadership_germany

Danarca committed:
German Revolution > German Revolt

Danarca committed:
A few comments

Danarca committed:
No any_country in every_country

Danarca committed:
German form AI willingness localisation + leftover edit

Danarca committed:
Simple Austrian shatter loc on the shatter_austria effect

Danarca committed:
Dont consider major powers opinions if you are one yourself

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1921 @@
+			trigger = {
+				this ?= c:AUS
+			}
+			add = 100
```
All these adds should be standardized into values. So you can tweak them.

Danarca replied:
Good point! I also have an inkling that you cannot use anything but `add` in button modifiers, so I've replaced the multiplies as well.

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1921 @@
+	visible = {
+		OR = {
+			var:sway_spending_level ?= 4
+			var:sway_menu_level ?= 1
```
Whatever this is, this is not a menu "level", and needs a better name lol. Maybe page.

Danarca replied:
Done

HiddeLekanne commented on common/scripted_buttons/imperia_frankfurt_parliament_buttons.txt:
```diff
@@ -0,0 +1,1921 @@
+			set_variable = {
+				name = sway_spending_amount
+				value = 50
+			}
```
Why not directly setting the amount instead of first the level then the amount.

Also these should be constants.

Danarca replied:
Done and done

Danarca committed:
Balancing

- Sway authority cost 15 -> 20
- Club influencing now cant go into minus, options wont be available
- 1 target cost 125 -> 75
- 3 targets cost 200 -> 300
- 5 targets cost 300 -> 550
- Gutter King / German Major personal sway cooldown 6 weeks ->  4 weeks
- German Prince sway cooldown from 3 months -> 6 weeks

HiddeLekanne reviewed (commented):
From yesterday, forgot to submit

Danarca committed:
sway_menu_level -> sway_menu_page

Danarca committed:
Standardised AI button willingness values into script values

Danarca committed:
Sway spend amounts set as values; gets initialised earlier

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+	}
+}
+
+imperia_frankfurt_parliament_gutter_king_consolidated = {
```
consolidate_gutter_king

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+	remove_variable ?= german_establishment_leader
+}
+
+imperia_frankfurt_parliament_shatter_austria = {
```
prone to desync?

Danarca replied:
Lou and I tested it, and it worked fine.

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+				has_law = var:gutter_king_old_lawgroup_governance_principles
+			}
+		}
+		activate_law = var:gutter_king_old_lawgroup_governance_principles
```
can't this just be activate_law ?=  var:gutter_king_old_lawgroup_governance_principles

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+						}
+					}
+				}
+				OR = {
```
Alsac Loraine?

Danarca replied:
If they have Alsace-Lorraine it would be ceded to the South Germans anyways.

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+	}
+}
+
+imperia_frankfurt_parliament_spawn_south_german_federation = {
```
I couldn't find where the laws of this new nation were set...?

Danarca replied:
The laws of it is inherited from the south_german_originator

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+	}
+}
+
+imperia_frankfurt_parliament_spawn_rhineland = {
```
I couldn't find where the laws of this new nation were set...?

Danarca replied:
Inherited from rhineland_originator

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+		name = frankfurt_parliament_right_faction_influence
+		value = 0
+	}
+	every_interest_group = {
```
jezus this is dynamic

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+	}
+}
+
+imperia_frankfurt_parliament_update_club_influence = {
```
This is one of the few functions you can actually make a lot shorter by giving variables, aka frankfurt_parliament_$FACTION$_faction_influence

HiddeLekanne commented on common/scripted_effects/imperia_frankfurt_parliament_effects.txt:
```diff
@@ -0,0 +1,1505 @@
+	}
+}
+
+imperia_frankfurt_parliament_BOOST_club = {
```
In fact all these 3 divided effects can be made shorter by giving the left, right or center as argument. Unfortunately you would need to make a second function which does the three calls.

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+		# Either theres more republic voters and monarchy voters
+		imperia_frankfurt_parliament_republic_voters_value > imperia_frankfurt_parliament_monarchy_voters_value
+		AND = {
+			# Or the two are equal
```
just call it a tie breaker.

HiddeLekanne replied:
not resolved

Danarca replied:
I'm confused, what change do you want to see on this one?

HiddeLekanne replied:
I meant just call it tiebreaker in the comments.

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+			NOT = {
+				imperia_frankfurt_parliament_oligarchy_voters_value < imperia_frankfurt_parliament_universal_voters_value
+			}
+			p:x11C0A0 = {
```
Tie breaker

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+		# Or oligarchy didnt lose, and Frankfurt voted for it
+		# (means a tie happened with at least one of them, possibly both)
+		AND = {
+			NOT = {
```
just do >= lol, no need for the not (not < is equal to >=)

Danarca replied:
Lol, good point! Dont know what I was thinking here

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+			NOT = {
+				imperia_frankfurt_parliament_wealth_voters_value < imperia_frankfurt_parliament_universal_voters_value
+			}
+			p:x11C0A0 = {
```
tie breaker

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+		# Or wealth didnt lose, and Frankfurt voted for it
+		# (means a tie happened with at least one of them, possibly both)
+		AND = {
+			NOT = {
```
just do >= lol, no need for the not (not < is equal to >=)

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+				owner = {
+					has_variable = frankfurt_parliament_voted_universal_suffrage
+				}
+			}
```
etc.....

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+			country_rank = rank_value:insignificant_power
+			root.country_rank <= rank_value:minor_power
+			# Special mention for Bavaria and Hanover, who didnt join the Frankfurt-led German empire...
+			trigger_if = {
+				limit = {
+					exists = c:BAV
+				}
+				root != c:BAV
+			}
+			trigger_if = {
+				limit = {
+					exists = c:HAN
+				}
+				root != c:HAN
+			}
```
Weird to hardcode these in like that.

Danarca replied:
Hanover and Bavaria were pretty much the only ones who didnt join the IRL Frankfurt Parliament government, so thats there for historicity.

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+				state_region = s:STATE_POSEN
+				state_region = s:STATE_SILESIA
+				state_region = s:STATE_WEST_PRUSSIA
+				state_region = s:STATE_EAST_PRUSSIA
```
Define these seperately, they have to be easily checked when an map update occurs. We need a file for region definitions.

Danarca replied:
I've added a new file; imperia_geography_triggers and put in the triggers for is_lesser_germany, is_german_confederation, is_rhineland and 2 for state is Frankfurt City and country owns Frankfurt city.

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+				state_region = s:STATE_STYRIA
+				state_region = s:STATE_AUSTRIA
+				state_region = s:STATE_MORAVIA
+				state_region = s:STATE_BOHEMIA
```
Define these separately, they have to be easily checked when an map update occurs. We need a file for region definitions.

HiddeLekanne commented on common/scripted_triggers/imperia_frankfurt_parliament_triggers.txt:
```diff
@@ -0,0 +1,621 @@
+				state_region = s:STATE_AUSTRIA
+				state_region = s:STATE_MORAVIA
+				state_region = s:STATE_BOHEMIA
+				state_region = s:STATE_ALSACE_LORRAINE
```
Define these separately, they have to be easily checked when an map update occurs. We need a file for region definitions.

HiddeLekanne reviewed (changes requested):
2nd partial review.

Danarca committed:
imperia_frankfurt_parliament_gutter_king_consolidated > imperia_frankfurt_parliament_consolidate_gutter_king

Danarca committed:
activate_law simplification

Danarca committed:
Tie-breakers

Danarca committed:
Imperia geography triggers (currently only for Frankfurt parliament)

Danarca committed:
A partly useful frankfurt_city trigger

Danarca committed:
Gutter King initial vote setup streamlining

Danarca committed:
Finishing up the Frankfurt City triggers

Danarca committed:
Forgot to save

Danarca committed:
More saving needed

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+	duration = 1
+	on_created_soundeffect = "event:/SFX/UI/Alerts/event_appear"
+	immediate = {
+		random_state = {
+			limit = {
+				imperia_is_frankfurt_city_state = yes
+			}
+			save_scope_as = frankfurt_location_scope
+		}
```
We couldn't make a script value named imperia_frankfurt_city_state?

Danarca replied:
I'm pretty sure script_values are just numbers. I'm not aware of them being able to be used as scopes

HiddeLekanne replied:
Yeay, it seems just a bit silly to go through every state to get this one.

HiddeLekanne replied:
So it would need to be an effect setting the scope, if anything.

Danarca replied:
> So it would need to be an effect setting the scope, if anything.

You just gave me an idea, what if I daisy-chained on the province ID, so, p:foobar.state and p:foobar.owner in the effect, perhaps?

HiddeLekanne replied:
don't know if that will work.

Danarca replied:
No it wont, just checked the actual use-cases of the current frankfurt_state triggers.

I mean.. We can do an on_monthly that updates a global variable and points it at Frankfurt state?

HiddeLekanne replied:
nah, not worth it I think

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+				}
+			}
+			else = {
+				# If this vote is tied, give it to the highest scored candidate, I guess
```
not highest scored, it just gives them in order of candidates, however they are generated (currently gdp)

Danarca replied:
I think that's what I meant when I wrote that comment, but honestly I'm a tiny bit confused, so clearly the comment didn't do its job properly.. Would need some extra love to make it the second-highest voted-for.

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+				}
+			}
+			scope:gutter_king_scope = {
+				trigger_event = {
```
ehm... why are we doing this?

Danarca replied:
Else the 0 - 100 meter that starts at 50 will *visually* be at 0 for everyone but the Gutter King.
The 50 bit is done in this event, so it starts in the middle for everyone :)

It was the only way I could see that worked, and I tried *a lot* to make it work :/

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+	duration = 2
+	on_created_soundeffect = "event:/SFX/UI/Alerts/event_appear"
+	immediate = {
+		random_country = {
+			limit = {
+				has_variable = accepted_gutter_crown
+			}
+			save_scope_as = gutter_king_scope
+		}
```
Is this needed?

Danarca replied:
It's for localisation reasons. So I'm pretty sure it's needed, unless it gets inherited from somewhere else

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+		remove_variable ?= frankfurt_parliament_supports_left_club
+		remove_variable ?= frankfurt_parliament_supports_center_club
+		remove_variable ?= frankfurt_parliament_supports_right_club
+		set_variable = {
+			name = club_change_cooldown
+			months = 4
+		}
```
Why is this not a modifier?

Danarca replied:
Because I hadn't thought of visualising cooldowns like that -.-
Want me to change it to a modifier?

HiddeLekanne replied:
In this case yes, there are a few others which should remain variables I thnk.

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+		has_variable = accepted_gutter_crown
+	}
+	immediate = {
+		# Poppet outside of the on_action to make it work properly
```
Popped*

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+					target = c:PRU
+				}
+			}
+			# Slight sidetrack to puppet Prussia, they dont get caught by the above loop for some reason...
```
german_establishment_aligned shouldn't contain prussia, unless they are still part of the conference, but don't accept the gutter crown?

Danarca replied:
I had an issue in a prepared test game where Prussia just wouldn't get caught by the `every_in_global_list` loop, that's why I gave them special treatment regarding the dominion'ing.

Which is fine, gave me an opportunity to make the Rhineland stuff dependent on them opposing the Gutter King!

HiddeLekanne commented on events/imperia_frankfurt_parliament_events.txt:
```diff
@@ -0,0 +1,1105 @@
+			variable = gutter_king_aligned
+			random_diplomatic_play = {
+				limit = {
+					has_variable = gutter_german_revolt_play
```
Can't you just use is_diplomatic_play_type?

Danarca replied:
Doh, yes I can. The variable bit is from when I first started on this and hadn't made a custom dp_type yet.
I'll take a look to see if that variable is needed for anything else.

HiddeLekanne commented on common/scripted_triggers/imperia_geography_triggers.txt:
```diff
@@ -0,0 +1,77 @@
+	}
+}
+
+imperia_country_owns_frankfurt_city = {
```
I would rather see a scripted value (if that is possible)

Danarca committed:
gutter_german_revolt_play variable > is_diplomatic_play_type

Danarca committed:
Typo in comments xp

Danarca committed:
club_change_cooldown | variable > modifier

Danarca committed:
Tiebreaker commentary

HiddeLekanne reviewed (approved):
Nothing left to say xd
